website/content/api-docs/secret/terraform.mdx

```diff
@@ -0,0 +1,310 @@
+Terraform Cloud tokens.
+
+| Method | Path                   |
+| :----- | :--------------------- |
+| `POST` | `/terraform/config` |
+
+### Parameters
+
+- `address` `(string: "")` – Specifies the address of the Terraform Cloud
+  server, if using Terraform Enterprise, provided as `"protocol://host:port"`.
+  The default is `https://app.terraform.io` for Terraform Cloud.
+
+- `token` `(string: "")` – Specifies the Terraform Cloud authentication token to
+  use. This token must have the needed permissions to manage all Organization,
+  Team, and User tokens desired for this mount.
+
+
+### Sample Payload
+
+```json
+{
+  "address": "https://app.terraform.io",
+  "token": "glhf..."
+}
+```
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --request POST \
+    --header "X-Vault-Token: ..." \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/terraform/config
+```
+
+## Read Access Configuration
+
+This endpoint queries for information about the Terraform Cloud connection.
+
+| Method | Path                   |
+| :----- | :--------------------- |
+| `GET`  | `/terraform/config` |
+
+### Sample Request
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    http://127.0.0.1:8200/v1/terraform/config
+```
+
+### Sample Response
+
+```json
+  "data": {
+    "address": "https://app.terraform.io",
+    "base_path": "/api/v2/"
+  }
+```
+
+
+## Create/Update Role
+
+This endpoint creates or updates the Terraform Cloud role definition in Vault.
+If the role does not exist, it will be created. If the role already exists, it
+will receive updated attributes. 
+
+Terraform Cloud offers three distinct types of API tokens with varying level of
+access: Organizations, Teams, and Users. A Vault Role can manage a single type
+of API token at a time, determined by how it is configured: 
+
+- To manage an Organization API token, provide the organization
+name with the `organization` parameter
+- To manage a Team API token, provide the `team_id` parameter
+- To manage a User API token, provide a `user_id` parameter
+
+~> **Special Note:** both Organizations and Teams can only have a single active API
+token at any given time. Generating a new token for a team or organization will
+effectively revoke any existing tokens, regardless if those tokens were
+originally created by Vault. As such, requesting an Organization or Team token
+from Vault will return the same token indefinitely until that token is rotated
+with the `/rotate-role` endpoint.
+
+Please see the [Terraform Cloud API
+Token documentation for more
+information](https://www.terraform.io/docs/cloud/users-teams-organizations/api-tokens.html).
+
+
+| Method | Path                |
+| :----- | :------------------ |
+| `POST` | `/terraform/role/:name` |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name for the Vault role. This is
+  part of the request URL.
+
+- `organization` `(string: "")` – Organization name to manage the single API
```

Comment:
Do organization and team_id not conflict with each other as well?

Reply:
They do not conflict. `team_id` takes precedence though